src/sdk/namespace_fs.js

```diff
@@ -66,6 +66,25 @@ function make_named_dirent(name) {
     return entry;
 }
 
+
+// this function will call native stat and check for matching uid, gid and `others` mode of a given account config
+// octal_others_mode < 4 is test is for read permission.
+// this number might be changed after supporting more backends types
+async function _stat_with_permissions_check(fs_account_config, stat_path) {
+    const res = await nb_native().fs.stat(fs_account_config, stat_path);
+    const octal_others_mode = parseInt(res.mode.toString(8), 10) % 10;
+    console.log('_stat_with_permissions_check', fs_account_config, res, octal_others_mode);
+    // handle root access
+    if (fs_account_config.uid === 0 && fs_account_config.gid === 0) return res;
+    if (res.uid !== fs_account_config.uid && res.gid !== fs_account_config.gid &&
```

Comment:
I don't follow - why are we checking for exact matching of uid and gid?
Why isn't the stat operation itself sufficient to fail with permission error if the file ACL is not allowing access?

Reply:
according to: https://man7.org/linux/man-pages/man2/lstat.2.html
`No permissions are required on the file itself` 
so for every uid gid it didn't fail, that's why I had to add these checks.

Reply:
ok, but checking for exact uid+gid match is not enough. We need to get the filesystem to check the permissions, because it might be using ACLs which we don't want to reimplement. 

Maybe we should https://man7.org/linux/man-pages/man2/access.2.html? 
Or perhaps instead of calling stat directly we should call open('r') and fstat() and then open for reading would fail?

WDYT @eshelmarc @motorman-ibm @nimrod-becker @dannyzaken

Reply:
The file system will check permission on each operation so as long as you set uid/gid the fs will do the rest

Reply:
fixed.